File: erettsegi_emelt/2010_may/java/Utas.java
public class Utas{
    
    public final int ules;
    public final int start;
    public final int end;
    public final int sorszam;
    
    public Utas(String line, int sorsz) {
        var split = line.split(" ");
        
        sorszam = sorsz;
        ules = Integer.parseInt(split[0]);
        start = Integer.parseInt(split[1]);
        end = Integer.parseInt(split[2]);
    }
    
    public int getTavolsag() {
        return end - start;
    }
    
    public int getAr(int kmAr) {
        var tav = getTavolsag();
        var utolso = tav % 10;
        var tizesek = tav / 10;
        
        if(utolso == 3 || utolso == 4 || utolso == 8 || utolso == 9) {
            ++tizesek;
        }
        
        return kmAr * tizesek;
    }
}

File: erettsegi_emelt/2010_may/c#/Utas.cs
public class Utas {

    public readonly int sorszam;
    public readonly int ules;
    public readonly int felszallasKm;
    public readonly int leszallasKm;

    public Utas(string line, int sorsz) {
        var split = line.Split(' ');

        sorszam = sorsz;
        ules = int.Parse(split[0]);
        felszallasKm = int.Parse(split[1]);
        leszallasKm = int.Parse(split[2]);
    }

    public static int getAr(Utas utas, int arPer10km) {
        var tav = utas.leszallasKm - utas.felszallasKm;
        var utolsoSzamjegy = tav % 10;
        var tizesek = tav / 10 + ((utolsoSzamjegy == 3 || utolsoSzamjegy == 4 || utolsoSzamjegy == 8 || utolsoSzamjegy == 9) ? 1 : 0);

        return arPer10km * tizesek;
    }
}